### PR TITLE
Adds a recipe for pymor

### PR DESCRIPTION
**Pull request by renefritze**

### CODE REVIEW

**Comment by renefritze** [comment]
Seems blocked by the same issue as #7658, @conda-forge/help-python

**Comment by renefritze** [comment]
I could use some help, @conda-forge/help-python
 - https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/22024293 is this in an error in the conda build of pyamg? I don't see an error from the actual test command for pymor.
 - https://circleci.com/gh/conda-forge/staged-recipes/34324?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link looks like an error with the pyopengl package itself?

**Comment by renefritze** [comment]
> have you tried adding a yum_requirements.txt with mesa-libGL-devel in it ?

That got me beyond the error indeed. Thanks @jschueller!

File: recipes/pymor/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+package:
+    name: pymor
+    version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/p/pymor/pymor-{{ version }}.tar.gz
+  sha256: ad5d0e8db09bf65580563632f5a73403790672f8ab33bc943da344fe70d290a0
+
+build:
+  number: 4
```
Comment (jschueller) [inline recipes/pymor/meta.yaml]: why 4 and not 0 ?

Comment (renefritze) [reply recipes/pymor/meta.yaml]: At some point I thought I had to manually bump that as counter. 

> The build number is used when the source code for the package has not changed but you need to make a new build. For example, if one of the dependencies of the package was not properly specified the first time you build a package, then when you fix the dependency and rebuild the package you should increase the build number.

Bit I guess that only refers to after the recipe has been accepted?

Comment (jschueller) [reply recipes/pymor/meta.yaml]: yup

**Comment by jschueller** [comment]
only the windows build to go, maybe pass "-v" to pytest to see what's happening ?

**Comment by renefritze** [comment]
> only the windows build to go, maybe pass "-v" to pytest to see what's happening ?

Oh, I misread the appveyor log then. I thought the pyamg stuff was already stopping the build. Trying now with verbose.

**Comment by jschueller** [comment]
maybe use `run: {{ pin_compatible('numpy') }}`, see https://conda-forge.org/docs/meta.html#building-against-numpy

**Comment by renefritze** [comment]
Right, thanks!

**Comment by renefritze** [comment]
This got me to a another new error on appveyor: https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/22142653#L1190
Can I somehow get to the actual Cython compile error? Or is the `error code 3221225477` telling me something I don't understand? Maybe @conda-forge/help-python-c can help?

The azure build for windows does not really start btw, which looks more like a general error than recipe related.

**Comment by jschueller** [comment]
it seems to indicate a crash, I wish to know what to do next

you can safely ignore the azure stuff for now

**Comment by renefritze** [comment]
So I've been trying a couple of things, and I've noticed that the order in which appveyor builds for python 3.6 and 3.7 for this recipe does not seem to be deterministic. Which is why I sometimes saw an error in the extension compile step and sometimes saw an error further along the process in testing. Only for different python versions. Now it looks like the python 3.6 build on appveyor is OK and tests succeed. The python 3.7 build however fails with an error I cannot see, and I don't see a way of forcing the msvc compiler step to show me the actual error. Unless ofc the numeric ID _actually_ is all that it emits, of which I cannot be sure. 

I really do not know how to proceed here, @conda-forge/help-python-c or @conda-forge/help-python

**Comment by jschueller** [comment]
Maybe you could skip py36 windows build temporarily to debug py37 ?

**Comment by renefritze** [comment]
Do you mean so the build fails faster by not doing everything for 36 that now succeeds? 
I

**Comment by jschueller** [comment]
yes, and not too depend on the random behavior

**Comment by renefritze** [comment]
I thought I did that in 
https://github.com/conda-forge/staged-recipes/pull/7669/commits/443cfbdf9697a85f0e66c4927c66752ef4bba123 just now, but the appveyor log looks like I failed: https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/22281016#L186

As a sidenote: is it correct that the it's always the miniconda3.6 install that's used, even or for py37?

**Comment by renefritze** [comment]
So it seems like I found a workaround for that weird build crash on windows. Does that look OK, @conda-forge/staged-recipes ?

File: recipes/pymor/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    - python
+    - ipython>=3.0
+    - ipyparallel
+    - matplotlib
+    - pyopengl
+    - pyamg  # [not win]
+    - pyqt
+    - pillow
+    - {{ pin_compatible('numpy') }}
+    - scipy>=0.13.3
```
Comment (scopatz) [inline recipes/pymor/meta.yaml]: There should be a space before the >

Comment (renefritze) [reply recipes/pymor/meta.yaml]: Fixed the other instances as well, thanks!

**Pull request merged**